tests/test_selfattention.py

```diff
@@ -175,6 +179,39 @@ def count_sablock_params(*args, **kwargs):
         nparams_default_more_heads = count_sablock_params(hidden_size=hidden_size, num_heads=num_heads * 2)
         self.assertEqual(nparams_default, nparams_default_more_heads)
 
+    @skipUnless(has_einops, "Requires einops")
+    def test_script(self):
+        for include_fc in [True, False]:
+            for use_combined_linear in [True, False]:
+                input_param = {
+                    "hidden_size": 360,
+                    "num_heads": 4,
+                    "dropout_rate": 0.0,
+                    "rel_pos_embedding": None,
+                    "input_size": (16, 32),
+                    "include_fc": include_fc,
+                    "use_combined_linear": use_combined_linear,
+                }
+        net = SABlock(**input_param)
+        input_shape = (2, 512, 360)
+        test_data = torch.randn(input_shape)
+        test_script_save(net, test_data)
+
+    @skipUnless(has_einops, "Requires einops")
+    @SkipIfBeforePyTorchVersion((2, 0))
+    def test_flash_attention(self):
```

Comment:
Have add test based on https://github.com/Project-MONAI/MONAI/pull/7977#discussion_r1701827839 for selfattention and cross attention.
Please help review, thanks! @mingxin-zheng @ericspod

Reply:
Similar idea to the previous comment, I would put the choice of `causal` into `parameterized.expand`, this also only tests with `causal==False` case as it currently is.

Reply:
Agree, good points, addressed in the latest commit.